### PR TITLE
[ADD] pos_check_session_state

### PR DESCRIPTION
**Pull request by legalsylvain**
Add a new simple module, that check that the PoS session is still in a opening state.
It prevents error, if a user close the session in a back office, and another waiter is using one.

**configuration**

pos_config_form

**Result**

error_message

### CODE REVIEW

**Review by chienandalu — CHANGES_REQUESTED** [review]
Nice module :)

Some remarks:

- As the pos window is blocked, it'd be nice to have a button in the dialog to close it:
image

- Config parameter could be in a more integrated style:
image

- Check lint errors

**Comment by legalsylvain** [comment]
Hi all.
Thanks for your review.
@MeetKD : I fixed your points.
@chienandalu : I refactored the code. (more simple, no more new Popup Object), and add a button OK. When clicked the PoS is closed that is the expected behaviour, because it is not possible to use it anymore.

could you update your review ? 

thanks !

File: pos_check_session_state/static/src/js/chrome.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                    clearInterval(self.intervalIDCheckSessionState);
+                }
+            })
+            .fail(function(error, event){
+                // Prevent error if server is unreachable
+                event.preventDefault();
+            });
+        }
+    });
+
```
Comment (MeetKD) [inline pos_check_session_state/static/src/js/chrome.js:70]: Can we remove extra line ?

File: pos_check_session_state/views/view_pos_config.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                        <label for="check_session_state_frequency"/>
+                        <div class="text-muted">
+                            The value is set in seconds.
+                        </div>
+                        <div class="content-group mt16">
+                            <field name="check_session_state_frequency" required="1"/>
+                        </div>
+                    </div>
+                </div>
+
```
Comment (MeetKD) [inline pos_check_session_state/views/view_pos_config.xml:26]: Can we remove extra line here too ?

**Review by MeetKD — CHANGES_REQUESTED** [review]

**Comment by legalsylvain** [comment]
@ MeetKD said:

> Can we remove extra line ?

I personally don't see why. Matter of taste.... But sure, feel free to make a PR, if you think it's important !

In the meantime, could you update your review ? 

Thanks !

**Review by MeetKD — APPROVED** [review]
@legalsylvain Yes its matter of taste :+1:  Updated Review for now.

**Comment by legalsylvain** [comment]
Hi @rafaelbn, could you unblock this PR ? I think i made the required changes.

thanks !

**Review by chienandalu — APPROVED** [review]

**Comment by chienandalu** [comment]
Sorry for the late review

/ocabot merge nobump

**Pull request merged**